cetmix_tower_git/models/cx_tower_server.py

```diff
@@ -82,20 +83,41 @@ def get_servers_by_git_ref(self, repository_url, head=None, head_type=None):
         recordset of cx.tower.server
             Matching servers. Empty recordset if no matches.
         """
+
+        server_obj = self.env["cx.tower.server"]
         # URL MUST be already canonical.
         if not repository_url:
-            return self.env["cx.tower.server"].browse()
+            return server_obj
 
-        Remote = self.env["cx.tower.git.remote"]
-        domain = [("url", "=", repository_url)]
+        # Get repository id by URL
+        repo_id = self.env["cx.tower.git.repo"]._get_repo_id_by_url(
+            repository_url, raise_if_invalid=True
+        )
+        if not repo_id:
+            return server_obj
+        repo = self.env["cx.tower.git.repo"].browse(repo_id)
+        if not repo.exists():
+            return server_obj
```

Comment:
If a record is found, there is no need to check `exists()` here, as the function `_get_repo_id_by_url` will always return an existing record.